plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -30,4 +30,16 @@ def valid_pipeline?(origin_id = id)
     end
     true
   end
+
+  # Make sure that this stage isn't referenced by another stage in a pipeline.
+  # This will stop soft_deletion of the stage.
+  def verify_not_part_of_pipeline
+    project.stages.each do |s|
+      if s.next_stage_ids.include?(id)
+        errors[:base] << "Stage #{name} is in a pipeline from #{s.name} and cannot be deleted"
+        return false
+      end
+    end
+    true
```

Comment:
True. Was just following the from of the validation method above, but this isn't a validation. Consider it gone.